README.md

```diff
@@ -131,83 +151,136 @@ const ipld = new Ipld({
 })
 ```
 
-##### `options.loadFormat(codec, callback)`
+##### `options.loadFormat(codec)`
 
 | Type | Default |
 |------|---------|
-| `Function` | `null` |
+| `async Function` | `null` |
 
-Function to dynamically load an [IPLD Format](https://github.com/ipld/interface-ipld-format). It is passed a string `codec`, the multicodec of the IPLD format to load and a callback function to call when the format has been loaded. e.g.
+Function to dynamically load an [IPLD Format](https://github.com/ipld/interface-ipld-format). It is passed a `codec`, the multicodec code of the IPLD format to load and returns an IPLD Format implementation. For example:
 
 ```js
+const multicodec = require('multicodec')
 const ipld = new Ipld({
-  loadFormat (codec, callback) {
-    if (codec === 'git-raw') {
-      callback(null, require('ipld-git'))
+  async loadFormat (codec) {
+    if (codec === multicodec.GIT_RAW) {
+      return require('ipld-git')
     } else {
-      callback(new Error('unable to load format ' + codec))
+      throw new Error('unable to load format ' + multicodec.print[codec])
     }
   }
 })
 ```
 
-### `.put(node, options, callback)`
+### `.put(node, format, options)`
+
+> Stores the given IPLD Node of a recognized IPLD Format.
+
+ - `node` (`Object`): the deserialized IPLD node that should be inserted.
+ - `format` (`multicodec`, required): the multicodec of the format that IPLD Node should be encoded in.
```

Comment:
This is general question / feedback:

I find passing name / code of the codec counter intuitive & I think just providing a codec instance might be better for following reasons:

1. Type checkers / linters  can catch mismatching / invalid references, but are useless with names.
2. If codec has not being installed you get an error, while with say `dag.put(node, dag.codec[name])` you are encourage to see if format is known.
3. Don't necessarily need to install codec to use it.

I understand that without installing a codec / format `decode` will not work, however `get` could optionally take codec as well. In case of `putMany / getMany` it's less clear but maybe optional codec dict can be provided as in `{[name]:Codec}`.

Not a big deal, just in my code I find it more convenient to pass actual codecs than names.

Reply:
I've opened https://github.com/ipld/js-ipld/issues/197 so that we don't lose that idea.